Files/UserControls/ModernNavigationToolbar.xaml.cs

```diff
@@ -234,30 +236,29 @@ public async void CheckPathInput(ItemViewModel instance, string CurrentInput)
                         : App.CurrentInstance.FilesystemViewModel.WorkingDirectory;
                     var parentItem = await StorageFileExtensions.GetFolderWithPathFromPathAsync(workingDir);
 
-                    if (CurrentInput.Contains("%temp%")) CurrentInput = CurrentInput.Replace("%temp%", AppSettings.TempPath);
-                    if (CurrentInput.Contains("%homepath%")) CurrentInput = CurrentInput.Replace("%homepath%", AppSettings.HomePath);
-                    CurrentInput = Environment.ExpandEnvironmentVariables(CurrentInput);
+                    currentInput = StorageFileExtensions.GetPathWithoutEnvironmentVariable(currentInput);
+                    if (currentSelectedPath == currentInput) return;
 
-                    var item = await DrivesManager.GetRootFromPath(CurrentInput);
+                    var item = await DrivesManager.GetRootFromPath(currentInput);
                     try
                     {
-                        var pathToNavigate = (await StorageFileExtensions.GetFolderFromPathAsync(CurrentInput, item, parentItem)).Path;
+                        var pathToNavigate = (await StorageFileExtensions.GetFolderFromPathAsync(currentInput, item, parentItem)).Path;
 
                         App.CurrentInstance.ContentFrame.Navigate(AppSettings.GetLayoutType(), pathToNavigate); // navigate to folder
                     }
                     catch (Exception) // Not a folder or inaccessible
                     {
                         try
                         {
-                            var pathToInvoke = (await StorageFileExtensions.GetFileFromPathAsync(CurrentInput, item, parentItem)).Path;
+                            var pathToInvoke = (await StorageFileExtensions.GetFileFromPathAsync(currentInput, item, parentItem)).Path;
```

Comment:
@soumyamahunt This should be
```cs
var pathToNavigate = (await StorageFileExtensions.GetFileWithPathFromPathAsync(currentInput, item, parentItem)).Path;
```
`GetFileFromPathAsync(...).Path` will be empty when browsing android/phones/mtp devices

Reply:
As you can see I didn't add or try to change this code, it was already there. My pr is not related to it in any way. Since I changed some lines before that it is showing as changed. If you want me to work on that issue, create a issue report on that and asign me to it.

Reply:
@soumyamahunt My bad I think this was changed in a previous PR also made by you